### PR TITLE
Fix migration auto-generation of Asset.blob

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: dandi/publish/storage.py [hunk 1]
```diff
@@ -2,11 +2,61 @@
...
+                    '%s.storage must be a subclass/instance of %s.%s'
+                    % (self.__class__.__qualname__, Storage.__module__, Storage.__qualname__)
+                )
+
+
+class DeconstructableFileField(CallableStorageFileField):
+    """
+    A FileField which preserves a callable "storage" argument during deconstruction.
+
+    This fixes a known bug in Django 3.1.
```
Comment (dchiquit) [inline dandi/publish/storage.py:30]: How do you feel about adding a link to the bug in the comment? I've found links useful if the comment was written as a reminder to remove a workaround so the developer doesn't have to go rooting around for the bug every time to check up on the status.

File: dandi/publish/storage.py [hunk 2]
```diff
@@ -2,11 +2,61 @@
...
 from django.conf import settings
 from django.core.files.storage import Storage, get_storage_class
+from django.db import models
 from minio_storage.policy import Policy
 from minio_storage.storage import MinioStorage, create_minio_client_from_settings
 from storages.backends.s3boto3 import S3Boto3Storage
 
 
+class CallableStorageFileField(models.FileField):
+    """A FileField backporting the Django 3.1 feature to allow a callable "storage" argument."""
```
Comment (dchiquit) [inline dandi/publish/storage.py:12]: Are you planning on removing this when we bump to 3.1?

File: dandi/publish/storage.py (hunk 1 above)
Comment (brianhelba) [reply dandi/publish/storage.py:30]: I haven't filed it yet, but I'm working on it getting it upstream to Django. They have a lot more red tape. :smile: 

There are a few other adjacent issues which I've also found (e.g. https://github.com/py-pa/django-minio-storage/pull/104 ), but those don't directly solve our problem.

File: dandi/publish/storage.py (hunk 2 above)
Comment (brianhelba) [reply dandi/publish/storage.py:12]: Yes.

File: dandi/publish/storage.py (hunk 1 above)
Comment (brianhelba) [reply dandi/publish/storage.py:30]: https://code.djangoproject.com/ticket/31941

**Pull request merged**